src/Symfony/Component/HttpFoundation/RedirectResponse.php

```diff
@@ -76,6 +83,10 @@ public function getTargetUrl()
      */
     public function setTargetUrl($url)
     {
+        if (!is_string($url)) {
```

Comment:
Cast on my side works fine. If we do not cast object then response contains valid body and invalid location header. This is the main argument to do this inside the function.